services/lobby.service.js

```diff
@@ -0,0 +1,1003 @@
+            await lobby.populate(['lobbyCreator', 'playerList']);
+            return lobby;
+        }
+
+    } catch (error) {
+        console.log(error);
+        throw error;
+    }
+};
+
+// join lobby with lobby code
+const joinLobby = async (req, res) => {
+    try {
+        const { lobbyCode, playerName } = req.body;
+
+        // Find the lobby with the given lobbyCode
+        let findLobby = await Lobby.findOne({ lobbyCode: lobbyCode });
+
+        if (findLobby) {
+
+            // Check if the lobby is already locked
+            if (findLobby.lobbyLocked) {
+                return Constants.LOBBY_LOCKED;
+            }
+
+            // Generate a random name for the user
+            // const randomName = await lobbyHelper.generateRandomName();
+
+            // Create a new user with the random name
+            const user = await User.create({ name: playerName });
+
+            // Add the new user to the playerList array in the lobby
+            const newUserList = [...findLobby.playerList, user._id];
+
+            // Update the lobby's playerList with the new user
+            const updatedLobby = await Lobby.findByIdAndUpdate(
+                findLobby._id,
+                { playerList: newUserList },
+                { new: true }
+            ).populate(['lobbyCreator', 'playerList']);
+
+            return updatedLobby;
+        }
+    } catch (error) {
+        console.log(error);
+    }
+};
+
+
+const playLobby = async (req, res) => {
+    try {
+        const { id } = req.body;
+
+        // Check if the user exists
+        const user = await User.findOne({ _id: id });
+        if (!user) {
+            return Constants.USER_NOT_FOUND;
+        }
+
+        // Check if the user has already created a lobby
+        const lobby = await Lobby.findOne({ lobbyCreator: id });
+        if (!lobby) {
+            return Constants.CREATE_LOBBY_FIRST;
+        }
+
+        // Check if the lobby has at least 3 players
+        if (lobby.playerList.length < 3) {
+            return Constants.MINIMUM_PLAYERS_REQUIRED;
+        }
+
+        if (lobby?.rounds[0]?.loop1?.length) {
+            return "mafi mil ski ha"
+        }
+
+        // Update the user's canPlayGame status
+        await User.findByIdAndUpdate(id, { canPlayGame: true }, { new: true });
+
+        // Lock the lobby and start the round
+        const updatedLobby = await Lobby.findByIdAndUpdate(
+            lobby._id.toString(),
+            { lobbyLocked: true },
+            { new: true }
+        );
+
+        // Start the round and get the updated lobby data
+        const result = await startRound(updatedLobby);
+        return {
+            code: 'success',
+            message: Constants.GAME_PLAYED_SUCCESSFULLY,
+            lobby: result,
+        };
+    } catch (error) {
+        console.log(error);
+        return Constants.INTERNAL_SEVER_ERROR;
+    }
+};
+
+
+
+const statusLobby = async (req, res) => {
```

Comment:
@komalsh09 what's the intended usage of `statusLobby`? From the code it looks like if status is true, we remove the player. Why is that? Maybe we need to rename the function?